Added Basalt and Humanize spack packages

iomaganaris opened the pull request:
Added Basalt and Humanize spack packages

- Needed to add Humanize package as well, as Basalt depends on it

iomaganaris committed:
Added Basalt and Humanize spack packages

matz-e commented on var/spack/repos/builtin/packages/basalt/package.py:
```diff
@@ -0,0 +1,45 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
```
From here to line 21 should be removed.

iomaganaris replied:
Yes, you are right :+1:

matz-e commented on var/spack/repos/builtin/packages/basalt/package.py:
```diff
@@ -0,0 +1,45 @@
+
+
+class Basalt(PythonPackage):
+    """C++11 Graph Storage library"""
```
But… it's a python package?

iomaganaris replied:
Yeeeap. Does this change something apart from the PythonPackage definition?

matz-e replied:
I was thinking that the comment seems to be a bit misleading? I'm basically wondering how to relate a `C+11 Graph Storage library` to Python. E.g. this would make it clearer to me:
```suggestion
    """Python interface of a C++ graph storage library"""
```

iomaganaris replied:
You have a point there. I think that a slight variation of the title you gave would be a little more representative.
```suggestion
    """C++11 Graph Storage library with Python interface"""
```
@tristan0x What do you think?

tristan0x replied:
That's good. Thank you

matz-e commented on var/spack/repos/builtin/packages/humanize/package.py:
```diff
@@ -0,0 +1,42 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
```
Remove instruction comment, too.

iomaganaris replied:
Same here, done.

matz-e commented on var/spack/repos/builtin/packages/humanize/package.py:
```diff
@@ -0,0 +1,42 @@
+    homepage = "https://github.com/jmoiron/humanize"
+    url      = "git@github.com:jmoiron/humanize.git"
+
+    # FIXME: Add proper versions and checksums here.
```
Address FIXMEs? Also, commented out code is dead code and should be removed IMO

iomaganaris replied:
Yes, I agree. I just forgot them :-1:

matz-e reviewed (changes requested):
Got some minor nitpicks, otherwise looks good! 🙃

tristan0x commented on var/spack/repos/builtin/packages/basalt/package.py:
```diff
@@ -0,0 +1,45 @@
+    version('develop', git=url, branch='master', submodules=True, preferred=True)
+    version('0.1.1', git=url, tag='v0.1.1', submodules=True)
+
+    depends_on('py-setuptools', type=('build', 'run'))
```
There are missing tests dependencies.

ferdonline replied:
Typically not a good idea to have develop version as preferred

matz-e replied:
test dependencies should be added with `when='+test'`, IIRC?

matz-e replied:
You can test this by building with `spack install --test=root`.

iomaganaris replied:
I created a new PR for just the humanize package. I also encapsulated the change suggested by @ferdonline . 
@tristan0x I did not find today the missing test dependencies. I will check again tomorrow. 
@matz-e Thanks for pointing that out.
And thank all of you for your comments.

iomaganaris replied:
Checked the tests that run correctly with the help of @matz-e :+1:

tristan0x reviewed (dismissed):
Would it be possible to split that change in 2 different revisions so that we can easily submit the `humanize` package upstream?

iomaganaris committed:
Modified basalt spack package.py and added py-humanize package.py

- Deleted commented lines of code in both of the packages
- Changed the basalt package discription
- Made version v0.1.1 as preferred in installation of basalt package
- Renamed humanize package to py-humanize package, as this is a python package
- Py-humanize package is the same added in the py-humanize PR

matz-e commented on var/spack/repos/builtin/packages/basalt/package.py:
```diff
@@ -0,0 +1,28 @@
+from spack import *
+
+
+class Basalt(PythonPackage):
```
BTW, Python packages are normally prefixed with `py-`. If there's a Basalt package for the C++ library, won't this lead to some conflict?

tristan0x replied:
Absolutely. This spack package should be named `py-basalt`.

iomaganaris replied:
You're right. I will take care of this.

iomaganaris committed:
Basalt python package renaming and tests checking

- Renamed the package to py-basalt as this is a python package
- Checked that the tests run correctly with
  'spack install --test root py-basalt'
- Added the ~mpi variant to py-h5py to stop tests from needing mpi

iomaganaris committed:
Deleted py-humanize from this PR and fixed a variant error in rocksdb package

pramodk reviewed (approved):
LGTM. As there are no additional dependencies in setup.py, approving.
